### PR TITLE
Fix Battle/Shell Armor + crit chance Tests

### PR DESCRIPTION
**Pull request by AlexOn1ine**
- Fixes Battle Armor and Shell Armor not blocking critical hits. 
- Removed `STATUS3_CANT_SCORE_A_CRIT` from `CalcCritChanceStageArgs` because it isn't used anywhere else but kept it in `constants/battle.h`. 
- Added some crit chance tests and moved `Dire Hit increases a battler's critical hit chance by 2 stages` to `crit_chance.c`. 
- Fix provided by MGriffin in `test_runner_battle.c` for `RNG_CRITICAL_HIT`.

### CODE REVIEW

File: test/battle/crit_chance.c [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+        TURN { MOVE(player, MOVE_LASER_FOCUS); }
+        TURN { MOVE(player, MOVE_TACKLE); }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_LASER_FOCUS, player);
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_TACKLE, player);
+        MESSAGE("A critical hit!");
+    }
+}
+
+SINGLE_BATTLE_TEST("If the target is poisoned the ability Merciless causes a moev to result in a critical hit")
```
Comment (DizzyEggg) [inline test/battle/crit_chance.c:121]: moev typo

File: test/battle/crit_chance.c [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+        OPPONENT(SPECIES_WOBBUFFET) { Status1(STATUS1_POISON); }
+    } WHEN {
+        TURN { MOVE(player, MOVE_TACKLE); }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_TACKLE, player);
+        MESSAGE("A critical hit!");
+    }
+}
+
+SINGLE_BATTLE_TEST("Focus Energy increases the user's critical hit ratio by one stage")
```
Comment (DizzyEggg) [inline test/battle/crit_chance.c:134]: I think Focus Energy increases the ratio by 2 stages?

**Review by DizzyEggg — CHANGES_REQUESTED** [review]
Looks good, solid work!

File: test/battle/crit_chance.c (hunk 2 above)
Comment (AlexOn1ine) [reply test/battle/crit_chance.c]: Yeah, wrong description. Corrections applied.

**Comment by AlexOn1ine** [comment]
Made a minor record ability change

File: src/battle_script_commands.c [hunk 3]
```diff
@@ -1914,14 +1914,13 @@ s32 CalcCritChanceStageArgs(u32 battlerAtk, u32 battlerDef, u32 move, bool32 rec
     s32 critChance = 0;
 
     if (gSideStatuses[battlerDef] & SIDE_STATUS_LUCKY_CHANT
-        || gStatuses3[battlerAtk] & STATUS3_CANT_SCORE_A_CRIT)
```
Comment (DizzyEggg) [inline src/battle_script_commands.c:1917]: Any reason you removed STATUS3_CANT_SCORE_A_CRIT? Could be useful if someone implemented a move which blocks critical hits.

Comment (AlexOn1ine) [reply src/battle_script_commands.c:1917]: Not real reason other then it isn't used anywhere else. I can add it back.

Comment (DizzyEggg) [reply src/battle_script_commands.c:1917]: I guess you can add it back for now, and later we'll decide what to do with that

Comment (AlexOn1ine) [reply src/battle_script_commands.c:1917]: added it back

**Pull request merged**